File: labs/solidworks/swex/add-in/documents-management/DocMgrAddIn.DocHandlerInit.vb
Private m_DocHandler As IDocumentsHandler(Of MyDocHandler)
Private m_DocHandlerGeneric As IDocumentsHandler(Of DocumentHandler)

Public Overrides Function OnConnect() As Boolean
    m_DocHandler = CreateDocumentsHandler(Of MyDocHandler)()
    m_DocHandlerGeneric = CreateDocumentsHandler()
    AddHandler m_DocHandlerGeneric.HandlerCreated, AddressOf OnHandlerCreated
    Return True
End Function

Private Sub OnHandlerCreated(ByVal doc As DocumentHandler)
    'TODO: implement
End Sub

File: labs/solidworks/swex/add-in/documents-management/DocMgrAddIn.DocHandlerInit.cs
private IDocumentsHandler<MyDocHandler> m_DocHandler;
private IDocumentsHandler<DocumentHandler> m_DocHandlerGeneric;

public override bool OnConnect()
{
    m_DocHandler = CreateDocumentsHandler<MyDocHandler>();
    m_DocHandlerGeneric = CreateDocumentsHandler();
    m_DocHandlerGeneric.HandlerCreated += OnHandlerCreated;
    return true;
}

private void OnHandlerCreated(DocumentHandler doc)
{
    //TODO: implement
}